components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2699,6 +2714,11 @@ protected void createAPI(API api) throws APIManagementException {
             String[] visibleRoles = new String[0];
             if (visibleRolesList != null) {
                 visibleRoles = visibleRolesList.split(",");
+                int index = 0;
+                for (String role: visibleRoles) {
+                    visibleRoles[index] = role.trim();
+                    index++;
+                }
```

Comment:
Same snippet repeated at many places. Can't we move this to a utility method and use elsewhere.